lib/sparrow/h2_worker/pool.ex

```diff
@@ -88,7 +96,7 @@ defmodule Sparrow.H2Worker.Pool do
           {:ok, pid}
   def start_link(config, pool_type, tags \\ []) do
     pool_name = config.pool_name
-    {:ok, pid} = start_link(config)
+    {:ok, pid} = start_link({config, pool_type, tags})
```

Comment:
Let's try to change arity of this function. If something breaks that *requires* arity 1, then fine, but otherwise there it just looks kinda weird.